src/Process/Runner/MutationTestingRunner.php

```diff
@@ -118,6 +127,13 @@ public function run(iterable $mutations, string $testFrameworkExtraOptions): voi
 
                 return true;
             })
+            ->filter(static function (Mutant $mutant) use ($notMatchedSourceCodeRegexes): bool {
+                if ($notMatchedSourceCodeRegexes !== []) {
+                    throw NotMatchedIgnoreSourceCodeRegexFound::forRegexes(array_keys($notMatchedSourceCodeRegexes));
+                }
+
+                return true;
+            })
```

Comment:
@sanmai I need your help here.

I feel like `sanmain/pipeline` needs one more method - `tap` (similar to this one https://lodash.com/docs/#tap)

What I want to do is after Mutants are filtered and we completed building `$notMatchedSourceCodeRegexes` - I would like to check it, but there is no possibility to do it since we have only `filter()` method on the `Standard` class, which **iterates over all the items again**.

So, desired logic:

```php
take($mutations)
    ->filter() // filter ignored mutants out and build $notMatchedSourceCodeRegexes
    // (!) new
    ->tap() // check $notMatchedSourceCodeRegexes and throw an exception if needed
    ->filter() // continue existing logic - filter not covered by tests mutants
```

do you have any recommendations here? is it the right way to go?

Reply:
The problem here is that there's no `array $mutants`; there are code paths when we produce them only as required. In other words, worst case there's `interable $mutants` with who knows how many values in it.

If you want to drop arbitrary parts of a sequence (say, first five, or two after ten), there's a handy `slice()` method, but I don't think that's what you're looking for here.

Reply:
I'm looking for a place (method) where I can do some job after the first `filter()` but before the second `filter()`

Reply:
Since you can't know if you're looking at the very last mutant, you can't throw an exception from anywhere here. Neither you should require to iterate over mutants all over again. Therefore I'd like to suggest to track regex usages, and only throw at some later stage. With that said I would much more prefer to see a notice: breaking non-PR builds can be a real pain.

There's another option: you can inject a fake mutant at the end of the stream, and then catch it somewhere else. I can elaborate further, but this will be incredibly scuffed and very opaque. Simple counting should work better.